Modules/Core/SpatialObjects/test/itkPolygonSpatialObjectIsInsideInObjectSpaceTest.cxx

```diff
@@ -0,0 +1,150 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkPolygonSpatialObject.h"
+#include <itkGroupSpatialObject.h>
+#include <itkSpatialObjectToImageFilter.h>
+#include <itkImageFileReader.h>
+#include <itkImageFileWriter.h>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+// -------------------------------------------------------------------------------------
+// The purpose of this test is to make sure that the function
+// ObjectIsInsideInObjectSpace is working as expected.
+// This is related to a regression that was fixed under case 1082
+// which showed that certain contours that were created from the polygonSpatialObject
+// had trailing tails that were not supposed to be there.
+// In order to test this we take a known regression list of contour points
+// and create a contour image from these points.
+// We compare the resulting image with the known baseline contour image.
+// ---------------------------------------------------------------------------------------
+
+using PointType = itk::ImageBase<2>::PointType;
```

Comment:
I think that even in tests we prefer to avoid global type aliases and template the helper functions over the necessary types.

Also, we would define the image dimensionality as a constant expression.

Reply:
Addressed.